src/Goerli-DssSpell.sol

```diff
@@ -41,29 +45,98 @@ contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
+        DssExecLib.setValue(MCD_END, "wait", EndAbstract(MCD_END_OLD).wait());
+        DssExecLib.setContract(MCD_END, "cure", MCD_CURE);
+        DssExecLib.authorize(MCD_CURE, MCD_END);
+
+        require(MCD_VAT == EndAbstract(MCD_END_OLD).vat(), "sanity-error-vat");
+        DssExecLib.setContract(MCD_END, "vat", MCD_VAT);
+        DssExecLib.authorize(MCD_VAT, MCD_END);
+        DssExecLib.deauthorize(MCD_VAT, MCD_END_OLD);
+
+        require(MCD_VOW == EndAbstract(MCD_END_OLD).vow(), "sanity-error-vow");
+        DssExecLib.setContract(MCD_END, "vow", MCD_VOW);
+        DssExecLib.authorize(MCD_VOW, MCD_END);
+        DssExecLib.deauthorize(MCD_VOW, MCD_END_OLD);
+
+        require(MCD_CAT == EndAbstract(MCD_END_OLD).cat(), "sanity-error-cat");
+        DssExecLib.setContract(MCD_END, "cat", MCD_CAT);
+        DssExecLib.authorize(MCD_CAT, MCD_END);
+        DssExecLib.deauthorize(MCD_CAT, MCD_END_OLD);
+
+        require(MCD_DOG == EndAbstract(MCD_END_OLD).dog(), "sanity-error-dog");
+        DssExecLib.setContract(MCD_END, "dog", MCD_DOG);
+        DssExecLib.authorize(MCD_DOG, MCD_END);
+        DssExecLib.deauthorize(MCD_DOG, MCD_END_OLD);
+
+        require(MCD_SPOT == EndAbstract(MCD_END_OLD).spot(), "sanity-error-spot");
+        DssExecLib.setContract(MCD_END, "spot", MCD_SPOT);
+        DssExecLib.authorize(MCD_SPOT, MCD_END);
+        DssExecLib.deauthorize(MCD_SPOT, MCD_END_OLD);
+
+        require(MCD_POT == EndAbstract(MCD_END_OLD).pot(), "sanity-error-pot");
+        DssExecLib.setContract(MCD_END, "pot", MCD_POT);
+        DssExecLib.authorize(MCD_POT, MCD_END);
+        DssExecLib.deauthorize(MCD_POT, MCD_END_OLD);
+
+        DssExecLib.setContract(MCD_ESM, "end", MCD_END);
+        DssExecLib.authorize(MCD_END, MCD_ESM);
+        DssExecLib.deauthorize(MCD_END_OLD, MCD_ESM);
+
+        bytes32[] memory ilks = IlkRegistryAbstract(ILK_REGISTRY).list();
+        for (uint256 i = 0; i < ilks.length; i++) {
+            bytes32 ilk = ilks[i];
+
+            address xlip = IlkRegistryAbstract(ILK_REGISTRY).xlip(ilk);
+            if (IlkRegistryAbstract(ILK_REGISTRY).class(ilks[i]) < 3) {
+                DssExecLib.deauthorize(xlip, MCD_END_OLD);
+                DssExecLib.authorize(xlip, MCD_END);
+            }
+
+            try DssExecLib.removeReaderFromWhitelist(IlkRegistryAbstract(ILK_REGISTRY).pip(ilk), MCD_END_OLD) {} catch {}
+            try DssExecLib.addReaderToWhitelist(IlkRegistryAbstract(ILK_REGISTRY).pip(ilk), MCD_END) {} catch {}
+        }
 
-        // MOMC Proposal
-        // https://vote.makerdao.com/polling/QmTmehbz#poll-detail
+        // --------------------------- New DssFlash ----------------------------
 
-        //Lower the WBTC-A Stability Fee from 3.25% to 2.25%.
-        DssExecLib.setIlkStabilityFee("WBTC-A", TWO_TWO_FIVE_PCT_RATE, true);
+        DssExecLib.authorize(MCD_VAT, MCD_FLASH);
+        uint256 newMax = FlashAbstract(MCD_FLASH_LEGACY).max() / 2;
```

Comment:
I assume 500M is not a hard limit, as someone will actually be able to flash loan 750M by loaning 500M on the legacy contract, then excuting the spell, then loaning 250M on the new contract.
Almost sure it is fine as I think 500M was set a while back in a quite arbitrary manner, but please confirm :).

Reply:
Good catch of this edge case. Yeah the 500M is quite arbitrary so I think we can leave with this for this unique possible occasion.